### ISSUE TITLE
moveViewToX problem

### ISSUE THREAD

**Issue by Begoodi** [danielgindi/ios-charts#450]
I have a bar chart with 100 items. I only display 10 at a time, and I enabled scrolling.
Everything is fine, but I would like to display the end of my graph instead of the beginning (fresh datas).
The only way to do it is to use the moveViewToX function in viewDidAppear.

The problem is that the graph is first displayed and animated at the beginning, then goes to the end, which is not very beautiful.
How can I do to display directly the graph at the right position ?

(When I move the moveViewToX function elsewhere than in viewDidAppear it does not work).

Thanks :)

**Comment by Begoodi** [comment]
Nobody has an idea ?
Perhaps my question is not clear, in fact I just want to display items 90 to 100 of my 100 items array...

**Comment by liuxuan30** [comment]
Well you opened another on SO: http://stackoverflow.com/questions/33038795/display-last-items-of-a-bar-chart-on-ios-charts/33049374#33049374

I am guessing because you are putting moveViewToX in viewDidAppear, it is in async mode with the chart rendering, so you cannot control it.

To answer your question, I cannot give you the exact solutions, but I can offer some ways you can try it:

subclass the chart view, call moveViewToX() somewhere in the chart view code, like the last line of drawRect(), to see if it works;

If you are familiar with ChartTransformer, there is a method called refresh in ChartViewPortHandler, it should be able to navigate to anywhere as long as you know how to play with the matrix. Check out this post:https://github.com/danielgindi/ios-charts/issues/226

**Comment by danielgindi** [comment]
Actually it should work, as it is deferred already to the rendering code.
But you should call it only after data is calculated (notifyDataSetChanged)

‏בתאריך יום שבת, 10 באוקטובר 2015, Xuan <notifications@github.com> כתב:

**Comment by Begoodi** [comment]
Hi Daniel,
Sorry but I don't understand your solution.
The moveViewToX function works properly as I can use it to go to the end of my bargraph and see last values.
What I would like is to display directly the graph in this position, today I'm obliged to display first values (1 to 10) an then go to values 90-100, and it makes the graph blink.
How to display values 90-100 in my example directly ?

**Comment by liuxuan30** [comment]
@begoodi what Daniel means is that call moveViewToX after you set your chart data.
Have you tried it, plus my suggestions?

**Comment by Begoodi** [comment]
Hi Liuxuan30, thanks for your answer.

My purpose is to directly show 90-100 when the chart pop up.

Calling moveViewToX after setting the data does not work, it still shows 1-10, and if I have to touch the barograph to display 90-100 (why I don't know).

I tried to subclass and use it in drawRect(), but didn't succeed, had lots of different errors. It must be my fault, not newbie, bur nearly ;)

I found a workaround by putting "barChartView.hidden = true" after calculations in viewDidLoad(), then in viewDidAppear() I use the moveViewToX function and then put "barChartView.hidden = true".
In that case I don't have the blink problem, I directly see 90-100 values.

The drawback is that I can't use the animation functions, because animations are done on values 1-10, and I hided this part of the display...
